### PR TITLE
Add more test coverage around fog transitions

### PR DESCRIPTION
**Pull request by karimnaaji**
Adds a few unit tests around untested fog transition code.

### CODE REVIEW

**Comment by rreusser** [comment]
Will review. While we're talking about fog tests, a related suspicion which I just discovered and haven't tracked in GH yet:

Our render tests might be snapshotting the image *before* fog elevation sampling has taken effect, so that the elevation-dependent render tests don't accurately reflect the final state of the idled map.

Screen Shot 2021-06-16 at 4 19 21 PM

**Comment by karimnaaji** [comment]
>Our render tests might be snapshotting the image before fog elevation sampling has taken effect, so that the elevation-dependent render tests don't accurately reflect the final state of the idled map.

Interesting, maybe the sampling should not be bound to a timer in the render test environment. Did you notice tests ever failing because of that?

**Comment by rreusser** [comment]
> Did you notice tests ever failing because of that?

I think the expectation and the result both fail to wait for the full transition so that the test doesn't actually fail. It's my interactive page that actually just runs the test as an interactive map that highlights the difference.

File: test/unit/ui/map.test.js [hunk 1]
```diff
@@ -347,6 +349,65 @@ test('Map', (t) => {
...
+                t.deepEqual(fog.properties.get('horizon-blend'), 0.5);
+
+                t.end();
+            });
+
+            t.test('fog respects validation option', (t) => {
+                const fog = new Fog({});
+                const fogSpy = t.spy(fog, '_validate');
+
+                fog.set({color: [444]}, {validate: false});
```
Comment (rreusser) [inline test/unit/ui/map.test.js:400]: Is 444 the color `0x1bc` in decimal?

**Review by rreusser — APPROVED** [review]
LGTM, nice addition 👍

File: test/unit/ui/map.test.js (hunk 1 above)
Comment (karimnaaji) [reply test/unit/ui/map.test.js:400]: Just a random value, nothing specific or meaningful there

**Pull request merged**